exporter/kafkaexporter/marshaler_test.go

```diff
@@ -147,3 +150,48 @@ func TestOTLPTracesJsonMarshaling(t *testing.T) {
 
 	assert.Equal(t, expectedJSON, final, "Must match the expected value")
 }
+
+type mockTraceMarshaler struct {
```

Comment:
You can simplify this by having `MockMarshaller` implement the field for all three signals. instead of a new type for each.

Reply:
I'm new to Go, but I don't think I can overload the Marshal() function for the respective interfaces.  I think I would have to change the *Marshaler interfaces to have differing function names.

Reply:
Aaa, no you're right. I did miss that.
Ignore me on this.